### PR TITLE
Fixes robocontrol on the 'puter

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

because you don't have an ID on you, it is never scanned for Robocontrol access, so you can just never use it unless you're on a tablet.

## Why It's Good For The Game

Bug fix,

## Changelog

:cl:
fix: Robocontrol on computers now work again.
/:cl:

### CODE REVIEW

**Comment by zxaber** [comment]
Having to insert your ID into modPC devices for them to read it was an intentional change.

All round-start modPC consoles have a card reader built in. If any are missing, it's an oversight.

**Comment by JohnFulpWillard** [comment]
> Having to insert your ID into modPC devices for them to read it was an intentional change.

And the problem isn’t that you not having an ID on you while your ID is in the PC means you can’t access any bots so cant do anything.

**Comment by zxaber** [comment]
Okay, I misread what the changes did earlier, I get it now. Good fix.

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 1]
```diff
@@ -203,21 +203,25 @@
...
 		return TRUE
 	if(!(bot_cover_flags & BOT_COVER_LOCKED)) // Unlocked.
 		return TRUE
 
-	var/obj/item/card/id/id_card = user.get_idcard(TRUE)
-	if(!id_card || !id_card.access)
+	var/obj/item/card/id/used_id
+	used_id = user.get_idcard(TRUE)
+	if(id) //IDs used directly overwrites
+		used_id = id
```
Comment (MrMelbert) [inline code/modules/mob/living/simple_animal/bot/bot.dm:217]:
```suggestion
	var/obj/item/card/id/used_id = id || user.get_idcard(TRUE)
```

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 2]
```diff
@@ -203,21 +203,25 @@
...
+
+	if(!used_id || !used_id.access)
 		return FALSE
-	id_card = id_card.GetID()
+	used_id = used_id.GetID()
```
Comment (MrMelbert) [inline code/modules/mob/living/simple_animal/bot/bot.dm:221]: why does it even do this ❓

Comment (JohnFulpWillard) [reply code/modules/mob/living/simple_animal/bot/bot.dm:218]: The original intent was to work with PDA's/tablets but the code above already does so... oops.

**Pull request merged**